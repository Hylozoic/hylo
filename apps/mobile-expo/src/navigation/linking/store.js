import { create } from 'zustand'

const useLinkingStore = create((set) => ({
  initialURL: null,
  setInitialURL: (url) => set({ initialURL: url }),
  returnToOnAuthPath: null,
  setReturnToOnAuthPath: (path) => set({ returnToOnAuthPath: path })
}))

export default useLinkingStore