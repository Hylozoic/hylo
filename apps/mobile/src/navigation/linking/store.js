import { create } from 'zustand'

const useLinkingStore = create((set) => ({
  initialURL: null,
  returnToOnAuthPath: null,
  setInitialURL: initialURL => set({ initialURL }),
  setReturnToOnAuthPath: returnToOnAuthPath => set({ returnToOnAuthPath })
}))

export default useLinkingStore
