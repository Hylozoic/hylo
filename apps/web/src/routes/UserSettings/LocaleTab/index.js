import LocaleTab from './LocaleTab'

export default LocaleTab
