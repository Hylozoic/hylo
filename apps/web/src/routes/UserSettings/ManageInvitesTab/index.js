import ManageInvitesTab from './ManageInvitesTab'
export default ManageInvitesTab
