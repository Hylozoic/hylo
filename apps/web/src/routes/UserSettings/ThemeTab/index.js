import ThemeTab from './ThemeTab'

export default ThemeTab 