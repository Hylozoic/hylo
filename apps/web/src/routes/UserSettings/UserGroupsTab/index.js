import UserGroupsTab from './UserGroupsTab'
export default UserGroupsTab
