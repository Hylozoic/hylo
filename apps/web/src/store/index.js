import { createStore } from 'redux'
import createMiddleware from './middleware'
// import { createEnhancer, reducer as routerReducer } from 'redux-data-router'
import createRootReducer, { createCombinedReducers } from './reducers'
import router from '../router'

export function getEmptyState () {
  // const combinedReducers = createCombinedReducers(routerReducer)
  const combinedReducers = createCombinedReducers({})
  return combinedReducers({}, { type: '' })
}

const store = createStore(
  // createRootReducer(routerReducer),
  createRootReducer(null),
  getEmptyState(),
  // createMiddleware(createEnhancer(router))
  createMiddleware(null)
)

export default store
