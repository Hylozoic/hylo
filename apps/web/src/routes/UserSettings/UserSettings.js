import { isEmpty, get } from 'lodash/fp'
import React, { useEffect } from 'react'
import { useTranslation } from 'react-i18next'
import { useSelector, useDispatch } from 'react-redux'
import { useLocation } from 'react-router-dom'

import { PROJECT_CONTRIBUTIONS } from 'config/featureFlags'
import EditProfileTab from './EditProfileTab/EditProfileTab'
import UserGroupsTab from './UserGroupsTab/'
import BlockedUsersTab from './BlockedUsersTab'
import ManageInvitesTab from './ManageInvitesTab/'
import LocaleTab from './LocaleTab'
import NotificationSettingsTab from './NotificationSettingsTab'
import AccountSettingsTab from './AccountSettingsTab/AccountSettingsTab'
import PaymentSettingsTab from './PaymentSettingsTab/PaymentSettingsTab'
import SavedSearchesTab from './SavedSearchesTab/SavedSearchesTab'
import FullPageModal from 'routes/FullPageModal'
import deactivateMe from 'store/actions/deactivateMe'
import deleteMe from 'store/actions/deleteMe'
import fetchUserSettings from 'store/actions/fetchUserSettings'
import logout from 'store/actions/logout'
import unBlockUser from 'store/actions/unBlockUser'
import { FETCH_FOR_CURRENT_USER } from 'store/constants'
import getBlockedUsers from 'store/selectors/getBlockedUsers'
import getMe from 'store/selectors/getMe'
import getMyMemberships from 'store/selectors/getMyMemberships'
import getQuerystringParam from 'store/selectors/getQuerystringParam'
import {
  updateUserSettings,
  unlinkAccount,
  registerStripeAccount
} from './UserSettings.store'
import { fetchLocation } from 'components/LocationInput/LocationInput.store'
import { setConfirmBeforeClose } from '../FullPageModal/FullPageModal.store'

const UserSettings = () => {
  const { t } = useTranslation()
  const dispatch = useDispatch()
  const location = useLocation()

  const currentUser = useSelector(getMe)
  const blockedUsers = useSelector(getBlockedUsers)
  const memberships = useSelector(state => getMyMemberships(state).sort((a, b) => a.group.name.localeCompare(b.group.name)))
  const confirm = useSelector(state => get('FullPageModal.confirm', state))
  const fetchPending = useSelector(state => state.pending[FETCH_FOR_CURRENT_USER])
  const queryParams = {
    registered: getQuerystringParam('registered', location)
  }

  useEffect(() => {
    dispatch(fetchUserSettings())
  }, [])

  const setConfirm = newState => {
    if (newState === confirm) return
    return dispatch(setConfirmBeforeClose(newState))
  }

  const content = [
    {
      name: t('Edit Profile'),
      path: 'edit-profile',
      component: (
        <EditProfileTab
          currentUser={currentUser}
          updateUserSettings={(...args) => dispatch(updateUserSettings(...args))}
          loginWithService={() => {}} // This action is not provided in the original code
          unlinkAccount={(...args) => dispatch(unlinkAccount(...args))}
          setConfirm={setConfirm}
          fetchLocation={(...args) => dispatch(fetchLocation(...args))}
          fetchPending={fetchPending}
        />
      )
    },
    {
      name: t('Groups & Affiliations'),
      path: 'groups',
      component: <UserGroupsTab personId={currentUser.id} />
    },
    {
      name: t('Invites & Requests'),
      path: 'invitations',
      component: <ManageInvitesTab currentUser={currentUser} />
    },
    {
      name: t('Locale'),
      path: 'locale',
      component: <LocaleTab currentUser={currentUser} />
    },
    {
      name: t('Notifications'),
      path: 'notifications',
      component: (
        <NotificationSettingsTab
          currentUser={currentUser}
          memberships={memberships}
        />
      )
    },
    {
      name: t('Account'),
      path: 'account',
      component: (
        <AccountSettingsTab
          currentUser={currentUser}
          deactivateMe={(...args) => dispatch(deactivateMe(...args))}
          deleteMe={(...args) => dispatch(deleteMe(...args))}
          logout={(...args) => dispatch(logout(...args))}
          setConfirm={setConfirm}
          updateUserSettings={(...args) => dispatch(updateUserSettings(...args))}
        />
      )
    },
    {
      name: t('Saved Searches'),
      path: 'saved-searches',
      component: <SavedSearchesTab />
    }
  ]

  if (currentUser && !isEmpty(blockedUsers)) {
    content.push({
      name: t('Blocked Users'),
      path: 'blocked-users',
      component: (
        <BlockedUsersTab
          blockedUsers={blockedUsers}
          unBlockUser={(...args) => dispatch(unBlockUser(...args))}
          loading={fetchPending}
        />
      )
    })
  }

  if (currentUser && currentUser.hasFeature(PROJECT_CONTRIBUTIONS)) {
    content.push({
      name: t('Payment'),
      path: 'payment',
      component: (
        <PaymentSettingsTab
          currentUser={currentUser}
          updateUserSettings={(...args) => dispatch(updateUserSettings(...args))}
          setConfirm={setConfirm}
          queryParams={queryParams}
          registerStripeAccount={(...args) => dispatch(registerStripeAccount(...args))}
        />
      )
    })
  }

  return <FullPageModal content={content} />
}

export default UserSettings
