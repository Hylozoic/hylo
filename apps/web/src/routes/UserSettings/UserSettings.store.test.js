import { updateAllMemberships, registerStripeAccount } from './UserSettings.store'

describe('updateAllMemberships', () => {
  it('matches snapshot', () => {
    expect(updateAllMemberships({ sendEmail: true })).toMatchSnapshot()
  })
})

describe('registerStripeAccount', () => {
  it('matches snapshot', () => {
    expect(registerStripeAccount('anauthorizationcodexyz')).toMatchSnapshot()
  })
})
