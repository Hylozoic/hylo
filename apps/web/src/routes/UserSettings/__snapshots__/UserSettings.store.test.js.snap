// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registerStripeAccount matches snapshot 1`] = `
Object {
  "graphql": Object {
    "query": "mutation ($authorizationCode: String) {
        registerStripeAccount(authorizationCode: $authorizationCode) {
          success
        }
      }",
    "variables": Object {
      "authorizationCode": "anauthorizationcodexyz",
    },
  },
  "meta": Object {
    "authorizationCode": "anauthorizationcodexyz",
  },
  "type": "UserSettings/REGISTER_STRIPE_ACCOUNT",
}
`;

exports[`updateAllMemberships matches snapshot 1`] = `
Object {
  "graphql": Object {
    "query": "mutation UpdateAllMemberships ($data: MembershipInput) {
    updateAllMemberships(data: $data) {
      success
      error
    }
  }",
    "variables": Object {
      "data": Object {
        "settings": Object {
          "sendEmail": true,
        },
      },
    },
  },
  "meta": Object {
    "optimistic": true,
    "settings": Object {
      "sendEmail": true,
    },
  },
  "type": "UserSettings/UPDATE_ALL_MEMBERSHIP_SETTINGS",
}
`;
